cypress/integration/buy-something.spec.ts

```diff
@@ -2,25 +2,39 @@ import {
   LoginPage, ProductListPage, ShoppingCart, Checkout, CheckoutOverview, OrderSummary
 } from '../page';
 
-  describe('Buy a t-shirt', () => {
-    const loginPage: LoginPage = new LoginPage();
-    const productListPage: ProductListPage = new ProductListPage();
-    const shoppingCartPage: ShoppingCart = new ShoppingCart();
-    const checkout:Checkout = new Checkout();
-    const checkoutOverview: CheckoutOverview = new CheckoutOverview();
-    const orderSummary: OrderSummary = new OrderSummary();
+describe('Buy something', () => {
+  describe('given a user is login to Saucedemo', () => {
+    before(() => {
+      const loginPage: LoginPage = new LoginPage();
+      cy.visit('https://www.saucedemo.com/');
+      loginPage.loginUser();
+    })
 
-  it('Then should be bought something', () => {
-    cy.visit('https://www.saucedemo.com/');
-    cy.getCookies();
-    cy.clearCookies();
-    loginPage.loginUser();
-    productListPage.addToCart();
-    productListPage.goShoppingCart();
-    shoppingCartPage.goProceedCheckout();
-    checkout.ingresarDatosComprador();
-    checkout.continueShopping();
-    checkoutOverview.finishShopping();
-    orderSummary.getConfirmOrder().should('have.text', 'THANK YOU FOR YOUR ORDER');
-    });
-});
+    describe('when the user adds a backpack to the cart', () => {
+      before(() => {
+        const productListPage: ProductListPage = new ProductListPage();
+        productListPage.addToCart();
+        productListPage.goShoppingCart();
+      })
+
+      describe('and continue with the purchase process, entering his information', () => {
+        before(() => {
+          const shoppingCartPage: ShoppingCart = new ShoppingCart();
+          const checkout: Checkout = new Checkout();
+          const checkoutOverview: CheckoutOverview = new CheckoutOverview();
+          shoppingCartPage.goProceedCheckout();
+          checkout.EnterBuyerData();
+          checkout.continueShopping();
+          checkoutOverview.finishShopping();
+        })
+
+        describe('then the order is finished and confirmed', () => {
```

Comment:
```suggestion
        describe('then the order should be finished and confirmed', () => {
```
try to have asset step always with the key word `should`